composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/data/EventRepositoryImpl.kt

```diff
@@ -36,25 +39,49 @@ class EventRepositoryImpl(
 
             is TaskyResult.Success -> {
                 if (!response.data.userExists) {
-                    TaskyResult.Error(CommunicationError.UnknownError()) // TODO better error handling
+                    TaskyResult.Error(TaskyError.UnknownError()) // TODO better error handling
                 } else {
                     localDataSource.saveAttendee(response.data.attendee.asAttendeeEntity())
                     TaskyResult.Success(
                         Attendee(
                             userId = response.data.attendee.userId,
-                            name = response.data.attendee.name
+                            name = response.data.attendee.name,
+                            email = response.data.attendee.email
                         )
                     )
                 }
             }
         }
     }
 
-    override suspend fun createEvent(event: Event): TaskyEmptyResult<CommunicationError> =
-        remoteDataSource.createEvent(event.asCreateEventRequest()).onSuccess { data ->
-            localDataSource.saveEvent(data.asEventEntity())
-            data.attendees.forEach { attendee ->
-                localDataSource.saveAttendee(attendee.asAttendeeEntity())
-            }
+    override suspend fun createEvent(event: Event): TaskyEmptyResult<TaskyError> {
+        // TODO if hostId doesn't exist we need to logout before executing this action
+        val eventEntity = event.asEventEntity(localDataSource.getHostId() ?: "")
+        val attendeeEntities = event.attendees.map {
+            it.asAttendeeEntity(
+                eventId = eventEntity.id,
+                remindAt = eventEntity.remindAt
+            )
+        }
+        val saveEventResponse = localDataSource.saveEvent(eventEntity)
+        val saveAttendeesResponse = localDataSource.saveAttendees(attendeeEntities)
+
+        if (saveEventResponse.isError() || saveAttendeesResponse.isError()) {
+            localDataSource.removeEvent(eventEntity)
+            localDataSource.removeAttendeesByEventId(eventEntity.id)
+            return saveEventResponse.asEmptyDataResult()
+        }
+
+        return remoteDataSource.createEvent(
+            event.asCreateEventRequest(
+                eventId = eventEntity.id
+            )
+        ).onError {
+            localDataSource.saveEvent(eventEntity.copy(isSynced = false))
+            localDataSource.saveAttendees(attendeeEntities.map { it.copy(isSynced = false) })
+        }.onSuccess {
+            localDataSource.saveEvent(eventEntity.copy(isSynced = true))
+            localDataSource.saveAttendees(attendeeEntities.map { it.copy(isSynced = false) })
         }.asEmptyDataResult()
+    }
```

Comment:
Not sure if I mentioned this or if Philipp did already, but considering this is an offline-first application, does it not make more sense first to add this data to your local DB and then schedule an API call? 

What I mean is a flow similar to the following example:

```
override suspend fun createReminder(reminder: AgendaItem): EmptyResult<DataError> {
    val localResult = localDataSource.upsertAgendaItem(reminder)
    if (localResult !is Result.Success) {
        return localResult
    }

    val remoteResult = remoteDataSource.postReminder(reminder)

    if (remoteResult is Result.Error) {
        applicationScope.launch {
            syncScheduler.scheduleSync(
                type = SyncAgendaItemsScheduler.SyncType.CreateReminder(
                    reminder = reminder
                )
            )
        }.join()
        return Result.Success(Unit)
    }

    return remoteResult
}
```

Reply:
I am not sure if I understand completely what you mean here. So I am already storing the data first in the database with:

```kotlin
val saveEventResponse = localDataSource.saveEvent(eventEntity)
val saveAttendeesResponse = localDataSource.saveAttendees(attendeeEntities)
```

The saving that happens here is only changing the boolean flag if the record already exists in the database if it was successfully sent to the backend or not. So we are only updating existing content in those onSuccess and onError calls.